src/main/java/org/springframework/guice/module/SpringModule.java

```diff
@@ -109,21 +119,36 @@ private void bind(ConfigurableListableBeanFactory beanFactory) {
 				Provider<?> namedProvider = BeanFactoryProvider.named(beanFactory,
 						beanName, type);
 				if (!type.isInterface() && !ClassUtils.isCglibProxyClass(type)) {
-					bindConditionally(binder(), name, type, typeProvider, namedProvider);
+					bindConditionally(binder(), name, type, typeProvider, namedProvider, bindingAnnotation);
 				}
 				for (Class<?> iface : ClassUtils.getAllInterfacesForClass(type)) {
-					bindConditionally(binder(), name, iface, typeProvider, namedProvider);
+					bindConditionally(binder(), name, iface, typeProvider, namedProvider, bindingAnnotation);
 				}
 				for (Type iface : type.getGenericInterfaces()) {
-					bindConditionally(binder(), name, iface, typeProvider, namedProvider);
+					bindConditionally(binder(), name, iface, typeProvider, namedProvider, bindingAnnotation);
 				}
 			}
 		}
 	}
 
+	private Optional<Annotation> getAnnotationForBeanDefinition(BeanDefinition definition) {
+		if (definition instanceof AnnotatedBeanDefinition
+				&& ((AnnotatedBeanDefinition) definition).getFactoryMethodMetadata() != null
+				&& ((AnnotatedBeanDefinition) definition)
+						.getFactoryMethodMetadata() instanceof StandardMethodMetadata) {
+			StandardMethodMetadata metadata = (StandardMethodMetadata) ((AnnotatedBeanDefinition) definition)
```

Comment:
In some cases (I don't quite understand yet which) we get a MethodMetadataReadingVisitor instead of StandardMethodMetadata. Any suggestions on a better way to go about this? I want the list of annotations on the factory method.